matrix/matrix.py

```diff
@@ -0,0 +1,198 @@
+            return False
+
+        return True
+
+    @classmethod
+    def unity(cls, number_lines: int):
+        """Generates a matrix with a main diagonal of 1's, everything else
+        is 0's.
+        """
+        tuples_in_chars = [
+            list(str('1' + ('0' * i)).zfill(number_lines)[::-1]) for i in
+            range(number_lines)]
+
+        return cls(*tuple(
+            [tuple(map(int, tuple_chars)) for tuple_chars in tuples_in_chars]))
+
+    @classmethod
+    def ones(cls, number_lines):
+        """Generates a matrix full of 1's."""
+        return cls(*tuple(
+            [tuple(repeat(1, number_lines)) for _ in range(number_lines)]))
+
+    def __repr__(self):
+        """Returns a string presentation of the matrix instance."""
+        return f"({', '.join([str(line) for line in self._matrix])})"
+
+    @property
+    def tuples(self):
+        """Returns the matrix instance as a tuple of tuples."""
+        return self._matrix
+
+    @staticmethod
+    def _bool_operation(operation, data) -> bool:
+        """Checks boolean operation output.
+
+        Args:
+            operation (function): the boolean function to operate.
+            data (list): the data of the matrix (list of tuples).
+
+        Returns:
+            If the operation is 'equal', returns true if all items are equal
+            to their parallel items in other, else false. If operation is
+            'not equal', return true if one or more items are not equal to
+            their parallel item/s in other, else false.
+        """
+        if operation == operator.eq:
+            return all(
+                [item for line_data in data for item in line_data])
+
+        elif operation == operator.ne:
+            return any(
+                [item for line_data in data for item in line_data])
+
+    def _calculate_matrix_on_matrix(self, other, operation):
+        """Does mathematical operation on two matrices.
+
+        Args:
+            other (Matrix): the second matrix to do the operation on.
+            operation (function): the mathematical function to operate.
+
+        Returns:
+            List of tuples which contains the calculated matrix data.
+        """
+        return [tuple([operation(self.tuples[i][j], other.tuples[i][j]) for j
+                       in range(len(self.tuples))]) for i in range(len(
+                        self.tuples))]
+
+    def _calculate_scalar_on_matrix(self, other, operation):
+        """Does mathematical operation on scalar and matrix.
+
+        Args:
+            other (int or float): the scalar.
+            operation (function): the mathematical function to operate.
+
+        Returns:
+            List of tuples which contains the calculated matrix data.
+        """
+        return [tuple([operation(self.tuples[i][j], other) for j in range(
+            len(self.tuples))]) for i in range(len(self.tuples))]
+
+    def _operations_on_matrix(self, other, operation):
```

Comment:
This method tries to do too much
See how you can reuse actions and be more concrete with every operator